File: convert-exam.js
import fs from 'fs';
import path from 'path';

// Function to convert a single exam file
function convertExamFile(inputPath, outputPath) {
    // Read the file content
    const content = fs.readFileSync(inputPath, 'utf8');
    
    // Split content into questions
    const questions = content.split(/Question \d+/).filter(q => q.trim().length > 0);
    
    // Process each question
    let formattedQuestions = [];
    let questionCounter = 1;
    
    for (const question of questions) {
        // Skip metadata lines
        const lines = question.split('\n').filter(line => 
            !line.includes('Answer saved') && 
            !line.includes('Marked out of') && 
            !line.includes('Not flagged') &&
            !line.includes('Flag question') &&
            !line.includes('Question text') &&
            line.trim().length > 0
        );
        
        if (lines.length === 0) continue;
        
        // Extract question text (first line that's not an option)
        let questionText = '';
        let optionsStartIndex = 0;
        
        for (let i = 0; i < lines.length; i++) {
            const line = lines[i].trim();
            // Check if this line looks like an option (starts with a number followed by a period)
            if (/^\d+\./.test(line)) {
                optionsStartIndex = i;
                break;
            } else if (line.length > 0) {
                questionText += line + ' ';
            }
        }
        
        // Extract options
        let options = [];
        let correctAnswers = [];
        
        for (let i = optionsStartIndex; i < lines.length; i++) {
            const line = lines[i].trim();
            // Check if this line is an option
            const optionMatch = line.match(/^(\d+)\.\s*(.*)/);
            if (optionMatch) {
                const optionNumber = optionMatch[1];
                const optionText = optionMatch[2];
                options.push({
                    letter: String.fromCharCode(64 + parseInt(optionNumber)), // Convert 1->A, 2->B, etc.
                    text: optionText
                });
            }
        }
        
        // Determine if this is single or multiple choice
        let isMultipleChoice = false;
        if (questionText.includes('Choose two') || 
            questionText.includes('Choose three') || 
            questionText.includes('Choose four') ||
            questionText.includes('Select one or more')) {
            isMultipleChoice = true;
        }
        
        // Format the question in markdown
        let formattedQuestion = `### Question ${questionCounter}\n`;
        formattedQuestion += `${questionText.trim()}\n\n`;
        
        // Add options
        for (const option of options) {
            formattedQuestion += `${option.letter}. ${option.text}\n`;
        }
        
        formattedQuestion += '\n';
        formattedQuestions.push(formattedQuestion);
        questionCounter++;
    }
    
    // Write the formatted content to the output file
    const fileName = path.basename(inputPath, '.md');
    const formattedContent = `---
title: ${fileName}
---

# ${fileName}

${formattedQuestions.join('\n')}`;

    fs.writeFileSync(outputPath, formattedContent);
    console.log(`Converted ${inputPath} to ${outputPath}`);
}

// Process all .md files in the directory
const examDir = 'src/docs/exam/vmware01';
const files = fs.readdirSync(examDir);

files.forEach(file => {
    if (file.endsWith('.md')) {
        const inputPath = path.join(examDir, file);
        const outputPath = inputPath; // Overwrite the original file
        try {
            convertExamFile(inputPath, outputPath);
        } catch (error) {
            console.error(`Error converting ${file}:`, error.message);
        }
    }
});

console.log('Conversion complete!');

File: convert_exam.py
import os
import re

def convert_exam_file(input_path, output_path):
    print(f"Processing {input_path}")
    
    # Read the file content
    with open(input_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Split content into questions
    question_blocks = re.split(r'Question \d+', content)
    question_blocks = [q for q in question_blocks if q.strip() and len(q.strip()) > 50]
    
    # Process each question
    formatted_questions = []
    question_counter = 1
    
    for block in question_blocks:
        # Remove metadata lines
        lines = block.split('\n')
        clean_lines = []
        for line in lines:
            if not any(metadata in line for metadata in [
                'Answer saved', 'Marked out of', 'Not flagged', 
                'Flag question', 'Question text'
            ]):
                clean_lines.append(line)
        
        # Join lines and clean up
        clean_text = ' '.join(clean_lines).strip()
        clean_text = re.sub(r'\s+', ' ', clean_text)
        
        # Extract question text (before first option)
        option_match = re.search(r'\d+\.', clean_text)
        if not option_match:
            question_counter += 1
            continue
            
        question_text = clean_text[:option_match.start()].strip()
        
        # Extract options
        options_text = clean_text[option_match.start():]
        options = []
        option_pattern = r'(\d+)\.\s*([^0-9]*)'
        
        for match in re.finditer(option_pattern, options_text):
            option_number = int(match.group(1))
            option_text = match.group(2).strip()
            if option_text:
                option_letter = chr(64 + option_number)  # Convert 1->A, 2->B, etc.
                options.append(f"{option_letter}. {option_text}")
        
        # Format the question
        formatted_question = f"### Question {question_counter}\n"
        formatted_question += f"{question_text}\n\n"
        formatted_question += '\n'.join(options) + '\n'
        
        formatted_questions.append(formatted_question)
        question_counter += 1
    
    # Write output
    file_name = os.path.basename(input_path).replace('.md', '')
    formatted_content = f"""---
title: {file_name}
---

# {file_name}

""" + '\n'.join(formatted_questions)
    
    with open(output_path, 'w', encoding='utf-8') as f:
        f.write(formatted_content)
    
    print(f"Converted {input_path} to {output_path}")

# Process all .md files
exam_dir = 'src/docs/exam/vmware01'
for file in os.listdir(exam_dir):
    if file.endswith('.md'):
        input_path = os.path.join(exam_dir, file)
        output_path = input_path  # Overwrite original
        try:
            convert_exam_file(input_path, output_path)
        except Exception as e:
            print(f"Error converting {file}: {e}")

print("Conversion complete!")